frontend/src/hooks/useSignUp.js

```diff
@@ -36,9 +38,9 @@ const useSignUp = () => {
           setLoading(false);
         } else {
```

Comment:
I feel that the if else block here might not be necessary as the error message can be propagated from the backend. Any thoughts?

Reply:
I think it is okay. Error message from the backend might not be user friendly so it is okay to detect status and send appropriate message. On that note, the message could be better suited for the user as user might not know what "backend" is. For instance, the message could be "Encountered issues connecting to server. Check if your connection is stable?"